### DISCUSSION TITLE
how to calculate axial vibration?

### DISCUSSION THREAD

**Discussion by Pierre-zhu** [petrobras/ross#1092] [Q&A] [answered]
First of all, I want to thank all people involved in building up this fantastic open source tool for rotor dynamics. I am begginer ,I  want to calculate axial vibration's free modal. but I can find any discuss or case, who can help me?

**Comment by jguarato** [comment, marked as answer]
Hi @Pierre-zhu,

I'll try to help you.

Start by creating a rotor model without bearing elements. Ensure that the model has 6 degrees of freedom to properly account for axial vibrations. For instance, instead of using the `ShaftElement` class, use the `ShaftElement6DoF` class for shaft elements, and the `DiskElement6DoF` class for disks. Once your model is set up, you can use the `run_campbell` method to analyze the system's modes and identify the axial modes. Also check discussion #1041 to see if it helps you.

Make sure your version of the ROSS library is up-to-date, as newer versions may include improved features and bug fixes.

**Comment by Pierre-zhu** [comment]
I have another question : how to set up one earthing stiffness (thruter bearing) for one node?

**Reply by jguarato** [reply]
Hi @Pierre-zhu!

As far as I know, there is no specific model for the thrust bearing available in ROSS yet. You would need to use the generic `BearingElement` class. Some people are working on this model right now, but I don't know exactly when they will make it available for use in ROSS.
